server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java

```diff
@@ -97,10 +99,22 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory, Index
             case HYBRIDFS:
                 // Use Lucene defaults
                 final FSDirectory primaryDirectory = FSDirectory.open(location, lockFactory);
-                final Set<String> mmapExtensions = new HashSet<>(indexSettings.getValue(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS));
+                Set<String> nioExtensions = new HashSet<>(indexSettings.getValue(IndexModule.INDEX_STORE_HYBRID_NIO_EXTENSIONS));
+                final List<String> mmapExtensions = new ArrayList<>(indexSettings.getValue(IndexModule.INDEX_STORE_HYBRID_MMAP_EXTENSIONS));
+                if (nioExtensions.isEmpty()) {
+                    List<String> allExtensions = new ArrayList<>(INDEX_STORE_HYBRID_ALL_EXTENSIONS);
+                    allExtensions.removeAll(mmapExtensions);
+                    nioExtensions = new HashSet<>(allExtensions);
+                } else {
+                    if (!mmapExtensions.equals(List.of("nvd", "dvd", "tim", "tip", "dim", "kdd", "kdi", "cfs", "doc"))) {
```

Comment:
I think this is not the right place to perform the validation: each setting has the validation function where you could check if the other settings already defined.

Reply:
Thanks for pointing that out. I've added a validation function now.